mscore/instrwidget.cpp

```diff
@@ -590,85 +723,93 @@ void InstrumentsWidget::on_addButton_clicked()
+      for (QTreeWidgetItem* item : _partiturSelectedItems) {
+            
+            
+            if (_partiturSelectionType == CHILDREN_SELECTED) {
+                  QTreeWidgetItem* parent = item->parent();
                   if (parent->childCount() == 1) {
-                        partiturList->takeTopLevelItem(partiturList->indexOfTopLevelItem(parent));
-                        delete parent;
+                        // Since multiselect was implemented this can occur, keep the last remaing item, but remove the others.
+                        removeButton->setEnabled(false);
+                        return;
+                  }
+                  if (((StaffListItem*)item)->op() == ListItemOp::ADD) {
+                        if (parent->childCount() == 1) {
+                              partiturList->takeTopLevelItem(partiturList->indexOfTopLevelItem(parent));
+                              delete parent;
                         }
-                  else {
-                        parent->takeChild(parent->indexOfChild(item));
-                        delete item;
+                        else {
+                              parent->takeChild(parent->indexOfChild(item));
+                              delete item;
                         }
                   }
-            else {
-                  ((StaffListItem*)item)->setOp(ListItemOp::I_DELETE);
-                  item->setHidden(true);
-
-                  // check if a staff is linked to this staff
-
-                  int idx = parent->indexOfChild(item);
-                  StaffListItem* sli = static_cast<StaffListItem*>(parent->child(idx+1));
-                  if (sli) {
-                        StaffListItem* sli2 = static_cast<StaffListItem*>(parent->child(idx+2));
-                        if (sli->linked() && !(sli2 && sli2->linked())) {
-                              sli->setLinked(false);
-                              partiturList->update();
+                  else {
+                        ((StaffListItem*)item)->setOp(ListItemOp::I_DELETE);
+                        item->setHidden(true);
+                        
+                        // check if a staff is linked to this staff
+                        
+                        int idx = parent->indexOfChild(item);
+                        StaffListItem* sli = static_cast<StaffListItem*>(parent->child(idx+1));
+                        if (sli) {
+                              StaffListItem* sli2 = static_cast<StaffListItem*>(parent->child(idx+2));
+                              if (sli->linked() && !(sli2 && sli2->linked())) {
+                                    sli->setLinked(false);
+                                    partiturList->update();
                               }
                         }
                   }
-            static_cast<PartListItem*>(parent)->updateClefs();
-            partiturList->setCurrentItem(parent);
+                  static_cast<PartListItem*>(parent)->updateClefs();
```

Comment:
Another unneeded static cast? I haven't actually checked if we have a `toPartListItem` function , so if it doesn't exist, ignore all of this :)